### PR TITLE
DOC-1088: Bump up our own API in search results

### PR DESCRIPTION
**Pull request by freakwriter**
Summary of changes:

- Added search boost to the /api/ index page.
- Removed some language Vale didn't like.
- Updated API auth page to reflect current UI labels.
- Added references to API docs and API endpoint reference docs to the top of the /api/ page.

### CODE REVIEW

**Comment by freakwriter** [comment]
@nik8n can you give a quick review to this? [Linear issue for reference](https://linear.app/n8n/issue/DOC-1088/n8ns-own-api-is-not-discoverable-in-docs-search).

File: docs/api/index.md [hunk 1]
```diff
@@ -23,7 +25,7 @@ The API documentation assumes you are familiar with REST APIs. If you're not, th
 * [KnowledgeOwl's guide to working with APIs](https://support.knowledgeowl.com/help/working-with-apis){:target=_blank .external-link}: a basic introduction, including examples of how to call REST APIs.
 * [IBM Cloud Learn Hub - What is an Application Programming Interface (API)](https://www.ibm.com/cloud/learn/api){:target=_blank .external-link}: this gives a general, but technical, introduction to APIs.
 * [IBM Cloud Learn Hub - What is a REST API?](https://www.ibm.com/cloud/learn/rest-apis){:target=_blank .external-link}: more detailed information about REST APIs.
-* [MDN web docs - An overview of HTTP](https://developer.mozilla.org/en-US/docs/Web/HTTP/Overview){:target=_blank .external-link}: REST APIs work over HTTP, and use HTTP verbs, or methods, to indicate the type of action to perform.
+* [MDN web docs - An overview of HTTP](https://developer.mozilla.org/en-US/docs/Web/HTTP/Overview){:target=_blank .external-link}: REST APIs work over HTTP and use HTTP verbs, or methods, to identify the action to perform.
```
Comment (imchairmanm) [inline docs/api/index.md:28]: No strong feelings, but maybe "specify" is a bit more active if it doesn't run afoul of any style rules.

File: docs/integrations/builtin/core-nodes/n8n-nodes-base.n8n.md [hunk 2]
```diff
@@ -10,8 +10,10 @@ priority: medium
 
 A node to integrate with n8n itself. This node allows you to consume the [n8n API](/api/) in your workflows.
 
+Refer to the [n8n Rest API documentation](/api/) for more information on using the n8n API. Refer to [API endpoint reference](/api/api-reference/) for working with the API endpoints directly.
```
Comment (imchairmanm) [inline docs/integrations/builtin/core-nodes/n8n-nodes-base.n8n.md:13]:
```suggestion
Refer to the [n8n REST API documentation](/api/) for more information on using the n8n API. Refer to [API endpoint reference](/api/api-reference/) for working with the API endpoints directly.
```

For consistency with capitalization used on the linked page.

**Review by imchairmanm — APPROVED** [review]
I left a few comments I wasn't sure about, but approving so you can use your judgement on changing.

File: docs/api/index.md (hunk 1 above)
Comment (freakwriter) [reply docs/api/index.md]: Good call. "Specify" is allowed. :)

**Pull request merged**